packages/astro/src/core/compile/compile.ts

```diff
@@ -47,7 +47,7 @@ export async function compile({
 			resultScopedSlot: true,
 			transitionsAnimationURL: 'astro/components/viewtransitions.css',
 			annotateSourceFile:
-				!viteConfig.isProduction && astroConfig.devOverlay && astroConfig.devOverlay.enabled,
+				viteConfig.mode === "development" && astroConfig.devOverlay && astroConfig.devOverlay.enabled,
```

Comment:
<img width="488" alt="image" src="https://github.com/withastro/astro/assets/69170106/cc09d309-a91b-4f93-b83d-7c37336bf032">
Yes, in testing, which is neither production nor dev.

Reply:
I'm under the impression that "astro build" in testing CI should behave the same as it would under a normal production environment. Is that not the case?

Reply:
I can't reproduce it

Reply:
I think you'd need to set `process.env.NODE_ENV="production"` before calling Astro's `build()` API to make that work. It makes sense to me because it's how Vite deals with the `NODE_ENV` and what other framework maintainers requested.

But in Astro's case maybe we can change the `build()` API to auto set the `NODE_ENV` to `production`? That would be the more correct fix, but I'm not sure if it's safe. For instance, the user will be unable to call `build()` and build a development version. When `NODE_ENV=test` by default, we can't know for sure what the intent is.

Reply:
Do we need the indirection in the first place?

Could we instead get "isDev" explicitly from `createVite()` instead?

Reply:
I should clarify, `createVite()` knows the astro command being executed, unrelated to environments variables or vite's heuristics. Could we pass that into this plugin?